### PR TITLE
Add Host API CRUD

### PR DESCRIPTION
**Pull request by mauromedda**
Implements the CRUD Host API calls
Add a brand icon to the project (golang robot with AWX in the body)
Add Host API usage examples

### CODE REVIEW

File: awxtesting/mockserver/mockdata/groups.go [hunk 1]
```diff
@@ -10,7 +10,7 @@ var (
         "results": [
             {
                 "id": 21,
-                "type": "group",
+                "type": 0,
```
Comment (Colstuwjx) [inline awxtesting/mockserver/mockdata/groups.go]: I think it should be string type, need to confirm

File: groups_test.go [hunk 2]
```diff
@@ -11,7 +11,7 @@ func TestListGroups(t *testing.T) {
 		expectListGroupsResponse = []*Group{
 			{
 				ID:   21,
-				Type: "group",
+				Type: 0,
```
Comment (Colstuwjx) [inline groups_test.go]: the same case.

**Comment by Colstuwjx** [comment]
BTW, functions like `AssociateGroup`, `DisAssociateGroup` should also have its own testcase.

File: awxtesting/mockserver/mockdata/groups.go (hunk 1 above)
Comment (mauromedda) [reply awxtesting/mockserver/mockdata/groups.go]: me too

**Pull request merged**